examples/with-supabase/app/auth/sign-up/route.ts

```diff
@@ -0,0 +1,39 @@
+import { createRouteHandlerClient } from '@supabase/auth-helpers-nextjs'
+import { cookies } from 'next/headers'
+import { NextResponse } from 'next/server'
+
+export const dynamic = 'force-dynamic'
+
+export async function POST(request: Request) {
+  const requestUrl = new URL(request.url)
+  const formData = await request.formData()
+  const email = String(formData.get('email'))
+  const password = String(formData.get('password'))
+  const supabase = createRouteHandlerClient({ cookies })
+
+  const { error } = await supabase.auth.signUp({
+    email,
+    password,
+    options: {
+      emailRedirectTo: `${requestUrl.origin}/auth/callback`,
+    },
+  })
+
+  if (error) {
```

Comment:
Have you considered doing the next-auth style catch all routes pattern? Then you wouldn't need to manually define these four route handlers, but just one that scaffolds out the rest.

Reply:
I think the catch all route is a little harder to parse and modify. One of the goals of this template was to get people started with an authenticated app as quickly as possible, without abstracting away what is actually happening with Supabase. Having them as separate routes makes it really easy for someone new to Supabase, or early on in their journey with Next.js, to remove one of these routes, add a new one, or modify login to do something like OAuth or magic link.

Also, since this is a template, the user doesn't need to declare each of these routes manually 👍